### ISSUE TITLE
Support for GitPod

### ISSUE THREAD

**Issue by Haberkamp** [supabase/cli#268]
# Bug report

## Describe the bug

It seems that it's not possible to start supabase in a gitpod workspace

## To Reproduce

Steps to reproduce the behaviour, please provide code snippets or a repository:

1. Create a new gitpod workspace
2. Install supabase via homebrew
3. Run supabase init and after that supabase start

## Expected behaviour

All containers should start successfully without an error.

## Screenshots

![Bildschirmfoto 2022-05-17 um 14 19 26](https://user-images.githubusercontent.com/35109813/168809694-738f028f-3e41-40ca-8742-33f4636460a6.png)

## System information

- OS: Remote GitPod Server (probably a Linux machine)
- Browser (if applies) Not relevant
- Version of supabase-js: 1.34.0
- Version of supabase-cli: 0.26.0
- Version of Node.js: 16.15.0

**Comment by soedirgo** [comment]
I'm not familiar with GitPod, but can you do everything the Docker CLI can do inside it?

**Comment by Haberkamp** [comment]
Yes, I was able to pull a docker image for adminer (a tool for database management with a web ui) and I could reach the application. So you're able to use the Docker CLI in GitPod.

**Comment by sweatybridge** [comment]
> When working inside a GitPod instance you are already inside a docker container.

Running CLI in docker would require docker-in-docker support. I have not tested this myself but it appears to be supported on GitPod https://github.com/gitpod-io/gitpod/issues/52#issuecomment-738629624

**Comment by zivbk1** [comment]
I have a similar problem, but different errors when I run `supabase start`.

```
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: failed to Lchown "/usr/local/lib/node_modules/pm2/node_modules/nssocket/.npmignore" for UID 1389985163, GID 386085923 (try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid): lchown /usr/local/lib/node_modules/pm2/node_modules/nssocket/.npmignore: invalid argument
```

and

```
Error: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: failed to Lchown "/usr/local/lib/node_modules/pm2/node_modules/nssocket/.npmignore" for UID 1389985163, GID 386085923 (try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid): lchown /usr/local/lib/node_modules/pm2/node_modules/nssocket/.npmignore: invalid argument
Try rerunning the command with --debug to troubleshoot the error.
```

See the attached [log.txt](https://github.com/supabase/cli/files/11184499/log.txt) file.

**Comment by zivbk1** [comment]
@Haberkamp Did you run `supabase login` before init and start?

> ## To Reproduce
> Steps to reproduce the behaviour, please provide code snippets or a repository:
> 
> 1. Create a new gitpod workspace
> 2. Install supabase via homebrew
> 3. Run supabase init and after that supabase start
> 4. See error

**Comment by zlanich** [comment]
I'm getting exactly what @zivbk1 posted as well. I've seen related issues on the internet that suggest modifying `/etc/subuid` and `/etc/subgid`, but those are not writeable from the terminal in GitPod. I can't find much else on the internet regarding this issue.

**Comment by sweatybridge** [comment]
I managed to run `supabase start` on GitPod using CLI v1.50.8. Here are the fixes we made

1. bump storage-api to v0.37.4 which [avoids installing nssocket](https://github.com/supabase/storage-api/pull/312)
2. updates health check to [resolve localhost to ipv4](https://github.com/supabase/cli/commit/4864abdf1dbf0b2a5da5df9c5b06a9298aade060) address

```bash
gitpod /workspace/cli (main) $ go run . start
Setting up Edge Functions runtime...
Seeding data supabase/seed.sql...
Started supabase local development setup.

         API URL: http://localhost:54321
```

Please give it another try after updating and feel free to reopen if it doesn't work.

**Comment by zlanich** [comment]
@sweatybridge YES, just bumping `supabase` in `package.json` fixed everything. Thank you!
